activestorage/app/models/active_storage/blob.rb

```diff
@@ -88,8 +92,11 @@ def create_after_upload!(**arguments_for_create_and_upload) #:nodoc:
     # in order to produce the signed URL for uploading. This signed URL points to the key generated by the blob.
     # Once the form using the direct upload is submitted, the blob can be associated with the right record using
     # the signed ID.
-    def create_before_direct_upload!(filename:, byte_size:, checksum:, content_type: nil, metadata: nil, record: nil)
-      create! filename: filename, byte_size: byte_size, checksum: checksum, content_type: content_type, metadata: metadata
+    # Option <tt>:service_name</tt> should be provided when blob should be stored in the service different from the
+    # default one.
+    def create_before_direct_upload!(filename:, byte_size:, checksum:, content_type: nil, metadata: nil, service_name: nil, record: nil)
+      service_name ||= ActiveStorage::ServiceRegistry.name_of(service)
```

Comment:
Would an `after_initialize { self.service_name ||=  ActiveStorage::ServiceRegistry.name_of(service) }` work?

Reply:
I realized this case is only possible when alternative service is not configured for the attachment, so we can just take the default one from `Rails.configuration.active_storage.service` and get rid of `#name_of`.